code/modules/ghostroles/spawner/human/human.dm

```diff
@@ -137,6 +141,20 @@
 		age = rand(35, 50)
 	M.age = Clamp(age, 21, 65)
 
+	if(has_idris_account)
+		var/datum/money_account/money_account = SSeconomy.create_account(M.real_name, rand(idris_account_min, idris_account_max), null, FALSE)
+		if(M.mind)
+			var/remembered_info = ""
+			remembered_info += "<b>Your account number is:</b> #[money_account.account_number]<br>"
+			remembered_info += "<b>Your account pin is:</b> [money_account.remote_access_pin]<br>"
+			remembered_info += "<b>Your account funds are:</b> [money_account.money]电<br>"
+
+			if(money_account.transactions.len)
+				var/datum/transaction/transaction = money_account.transactions[1]
+				remembered_info += "<b>Your account was created:</b> [transaction.time], [transaction.date] at [transaction.source_terminal]<br>"
+			M.mind.store_memory(remembered_info)
+			M.mind.initial_account = money_account
```

Comment:
since you also use this in the other PR, this should probably be made into a proc, since now I think it's copypasted in 3 different locations

Reply:
i'll wait for the other one to get merged and then consolidate it in this one, then